build/mixxx.py

```diff
@@ -39,8 +39,8 @@ def __init__(self, target, machine, build, toolchain, available_features=[]):
         if build is None:
             build = 'debug'
 
-        if not build in ['debug', 'release']:
-            raise Exception("invalid build type")
+        if not build in ['debug', 'release', 'none']:
```

Comment:
I think the release version is picked in that case, since we always check only for the debug flag.
I do not bother much if a plain build=none gives not a sensible result, because that is not the use case for the option.
For me it is more the last resort to fix an exotic problem on a currently unsupported build platform without editing the scons scripts,  but passing magic flags on the command line.